src/pages/SignUp.js

```diff
@@ -0,0 +1,168 @@
+import { useState, useEffect } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import toast from 'react-hot-toast'
+import { useNavigate } from 'react-router-dom'
+import { signUp } from '../features/auth/authSlice'
+import { HeroImgContainer } from '../components'
+export const SignUp = () => {
+    const [formData, setFormData] = useState({
+        name: "", username: "", email: "", password: "", confirmPassword: ""
+    })
+    const { password, confirmPassword } = formData
+    const dispatch = useDispatch()
+    const navigate = useNavigate()
+    const { token } = useSelector((state) => state.auth)
+    const submitHandler = (e) => {
+        e.preventDefault()
+        if (password !== confirmPassword) {
+            toast.error('Passwords didn\'t not match')
+        }
+        else if (token && password === confirmPassword) {
+            dispatch(signUp(formData))
+            toast.success('Signed Up Successfully!', { duration: 1500, position: 'bottom-center' });
```

Comment:
Suggestion: just extract the repeated {duration: 1500, position: 'bottom-center'} in a utils function and reuse it.